legacy/toml-config/src/test/java/io/zeebe/legacy/tomlconfig/LegacyConfigurationSupportTest.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.0. You may not use this file
+ * except in compliance with the Zeebe Community License 1.0.
+ */
+package io.zeebe.legacy.tomlconfig;
+
+import io.zeebe.legacy.tomlconfig.LegacyConfigurationSupport.Replacement;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import org.assertj.core.api.Assertions;
+import org.junit.Test;
+
+public class LegacyConfigurationSupportTest {
```

Comment:
Should we have a test that the environment variables are properly mapped?

Reply:
I would love to. I just don't know how.

I could write a test that verifies the map entries. This would catch my own typos, but not whether I misunderstood something completely.

I would love to write an end-to-end test, but I don't know how to manipulate environment variables in a test - other than using Docker or something like this. Even if I managed to manipulate the environment settings, I wouldn't know how to observe whether the right configuration was set.

If there is something obvious I missed, please let me know.

Other then that I tested it manually - not it with all environment variables but with a few.

Reply:
Hm, you're right, the only thing I can come up with is a TestContainers setup - with actuators enabled we could get the config out, but I guess this is not possible yet. Let's leave it as is then :+1: